ledger-core/v2/runner/runner_rpc.go

```diff
@@ -6,22 +6,102 @@
 package runner
 
 import (
+	"fmt"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar"
+	"github.com/insolar/assured-ledger/ledger-core/v2/runner/executionevent"
 	"github.com/insolar/assured-ledger/ledger-core/v2/runner/executor/common/rpctypes"
 	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
 )
 
 func (r *DefaultService) GetCode(_ rpctypes.UpGetCodeReq, _ *rpctypes.UpGetCodeResp) error {
-	panic(throw.NotImplemented())
+	panic(throw.Unsupported())
 }
 
-func (r *DefaultService) RouteCall(_ rpctypes.UpRouteReq, _ *rpctypes.UpRouteResp) error {
-	panic(throw.NotImplemented())
+func (r *DefaultService) CallMethod(in rpctypes.UpRouteReq, out *rpctypes.UpRouteResp) error {
+	sink := r.getExecutionSink(in.ID)
+	if sink == nil {
+		panic("failed to find ExecutionContext")
+	}
+
+	event := executionevent.NewRPCBuilder(in.Request, in.Callee).
+		CallMethod(in.Object, in.Prototype, in.Method, in.Arguments).
+		SetImmutable(in.Immutable)
+	sink.ExternalCall(executionevent.RPC(event))
+
+	rawValue := <-sink.input
+
+	if in.Saga {
+		if rawValue == nil {
+			return nil
```

Comment:
could you explain this 'return' statement?

Reply:
deleted